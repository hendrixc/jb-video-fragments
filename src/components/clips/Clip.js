import React, { Component } from 'react';
import {
  Card,
  CardPrimaryAction,
  CardMedia,
} from '@rmwc/card';
import './clip.css';
import '@material/card/dist/mdc.card.css';
import '@material/button/dist/mdc.button.css';
import '@material/icon-button/dist/mdc.icon-button.css';
import PropTypes from 'prop-types';
import timeUtils from '../../utils/timeUtils';
import I18n from '../../i18n';

class Clip extends Component {
  render() {
    const { clip, onClick } = this.props;
    return (
      <div className="clip">
        <Card style={{ width: '15rem' }}>
          <CardPrimaryAction onClick={() => { onClick(clip); }}>
            <CardMedia
              style={{
                backgroundImage:
                  'url(/images/asche.jpg)',
              }}
            >
              <div>
                <p className="name ellipsis">
                  {clip.name}
                </p>
                <div className="start ellipsis">
                  {`${I18n.t('clip.start')}: ${timeUtils.formatSecondsToMSS(clip.start)}`}
                </div>
                <div className="end ellipsis">
                  {`${I18n.t('clip.end')}: ${timeUtils.formatSecondsToMSS(clip.end)}`}
                </div>
              </div>
            </CardMedia>
          </CardPrimaryAction>
        </Card>
      </div>
    );
  }
}

Clip.propTypes = {
  clip: PropTypes.object,
  onClick: PropTypes.func,
};

Clip.defaultProps = {
  clip: null,
  onClick: () => {},
};

export default Clip;
