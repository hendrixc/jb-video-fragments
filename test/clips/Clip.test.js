import React from 'react';
import renderer from 'react-test-renderer';
import Clip from '../../src/components/clips/Clip';

jest.mock('../../src/components/video/Video', () => 'Video');

it('renders an clip component into the Clip as snapshot', () => {
  const clip = {
    name: 'cualquier cosa',
    thumbnail: 'cualquier_cosa.png',
    tags: [],
    start: 80,
    end: 90,
  };
  const TextInputComponent = renderer.create(<Clip clip={clip} />).toJSON();
  expect(TextInputComponent).toMatchSnapshot();
});
