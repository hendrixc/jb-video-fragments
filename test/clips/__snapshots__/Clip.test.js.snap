// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an clip component into the Clip as snapshot 1`] = `
<div
  className="clip"
>
  <div
    className="mdc-card"
    style={
      Object {
        "width": "15rem",
      }
    }
  >
    <div
      className="mdc-card__primary-action"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <section
        className="mdc-card__media"
        style={
          Object {
            "backgroundImage": "url(/images/asche.jpg)",
          }
        }
      >
        <div>
          <p
            className="name ellipsis"
          >
            cualquier cosa
          </p>
          <div
            className="start ellipsis"
          >
            Start: 1:20
          </div>
          <div
            className="end ellipsis"
          >
            End: 1:30
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
